src/d/actor/d_a_sail.cpp

```diff
@@ -438,8 +456,39 @@ u8 l_AmastDL[] ALIGN_DECL(32) = {
 };
 
 /* 000000EC-00000364       .text setCorrectNrmAngle__15daSail_packet_cFsf */
-void daSail_packet_c::setCorrectNrmAngle(short, float) {
-    /* Nonmatching */
+void daSail_packet_c::setCorrectNrmAngle(s16 param_0, f32 param_1) {
+    s16 rnd = 900 + (s32)cM_rndF(200.0f);
+    this->m1C38 += rnd;
+
+    this->m1C34 = 300.0f * JMASSin(this->m1C38);
+
+    s16 r28 = param_0 + 0x8000;
+    s32 r27 = param_0;
+
+    s16 r26 = l_HIO.m10 * (1.0f - 0.5f * param_1);
+    s32 r0 = abs(r28);
+    s16 r30 = 1.25f * r26 * 182.04445f;
```

Comment:
do either of these inlines match for all the conversions here? would make it a bit more readable if so
```suggestion
    s16 r30 = 1.25f * cAngle::d2s(r26);
```
```suggestion
    s16 r30 = 1.25f * cM_deg2s(r26);
```